### DISCUSSION TITLE
Nuxt3 icon library

### DISCUSSION THREAD

**Discussion by abeldebruijn** [nuxt/nuxt#16001] [Questions] [answered]
I would like to implement icons in my Nuxt3 project. I have tried to use `@nuxtjs/fontawesome` or other libraries that are supported in Nuxt2. When I try to add a library to my `nuxt.config.js` file like underneath:
```js
import { defineNuxtConfig } from "nuxt3";
export default defineNuxtConfig({
  buildModules: [ "@nuxtjs/fontawesome"],
});
```

The console starts yelling at me that I'am doing something wrong
```js
__vite_ssr_import_0__.default.component is not a function
  at $id_0a4ce168 (file://./.nuxt/dist/server/server.mjs:2279:31)  
  at async __instantiateModule__ (file://./.nuxt/dist/server/server.mjs:5575:3)
__vite_ssr_import_0__.default.component is not a function
  at $id_0a4ce168 (file://./.nuxt/dist/server/server.mjs:2279:31)  
  at async __instantiateModule__ (file://./.nuxt/dist/server/server.mjs:5575:3)
```

What would be a viable way to implement icons in my Nuxt project? (All icon libraries are welcome)

**Comment by HealerNguyen** [comment]
for my use case fontawesome, you can using @fortawesome/fontawesome-free
- install it with npm or yarn 
- in nuxt.config.ts add this css to css: [ ] block like this:
 'import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
 css: [
     '~/assets/main.scss', // your different style
     '@fortawesome/fontawesome-free/css/all.min.css'
   ],
})
finally in .vue page add your icon<i class="fab fa-twitter"></i>

**Comment by HealerNguyen** [comment]
if you still stuck with font-awesome you can try this https://icones.js.org/

**Comment by abeldebruijn** [comment, marked as answer]
What I eventually did was importing `https://fonts.googleapis.com/css?family=Material+Icons|Material+Icons+Outlined`  as a meta link.

This is my `defineNuxtConfig` now:
```javascript
export default defineNuxtConfig({
  meta: {
    link: [
      {
        href: 'https://fonts.googleapis.com/css?family=Material+Icons|Material+Icons+Outlined',
        rel: 'stylesheet',
      },
    ],
  },
});
```

To use them in your component:
```html
<template>
  <i class="material-icons-outlined">{{ icon-name }}</i>
</template>
```

For more documentation: https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#meta

**Reply by itmysm** [reply]
can u show me a example? i getting this warning 
![deb](https://user-images.githubusercontent.com/75526205/171601738-51127691-3452-412f-80b3-f157e4837deb.png)

**Reply by solrobinson** [reply]
![image](https://user-images.githubusercontent.com/63585019/172022587-3b1830be-7060-4f88-8808-bef8c3c5c2f2.png)
![image](https://user-images.githubusercontent.com/63585019/172022596-61b97551-b373-4c54-8624-9f337919806b.png)

**Comment by lewebsimple** [comment]
If you'd like to use SVG icons (and thus benefit from tree-shaking), you may check out this [starter](https://github.com/viandwi24/nuxt3-awesome-starter) which uses `unplugin-icons`.

You may also check out this [module](https://github.com/lewebsimple/nuxt3-svg) to use `vite-svg-loader`, which allows to import SVG files as components directly. You can then add the Fontawesome svgs directory to auto-imported [components](https://v3.nuxtjs.org/guide/directory-structure/components/#library-authors) with the components option in nuxt.config.ts.

**Comment by cawa-93** [comment]
I just publish [iconify-prerendered](https://github.com/cawa-93/iconify-prerendered) — superset of icons. Each icon is standalone vue component, so you can use in easily without any plugins. Works offline. No external resourses needed. Zero-dependencies. 
```vue
<script setup>
import { IconAccount } from '@iconify-prerendered/vue-mdi' 
</script>

<template>
  <IconAccount/>
</template>
```

**Comment by FileSoft-us** [comment]
https://github.com/nuxt-modules/icon

**Reply by issamlk1** [reply]
even better than the selected answer (for my use case)
